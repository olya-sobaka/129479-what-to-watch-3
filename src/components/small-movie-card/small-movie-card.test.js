import React from "react";
import renderer from "react-test-renderer";
import SmallMovieCard from "./small-movie-card.js";

const movie = {
  title: `Kiki's Delivery Service`,
  posterSrc: `img/midnight-special.jpg`,
  id: 36,
  previewMp4: `https://coverr.co/videos/norway-road-f0J1etTrcQ`,
  previewWebm: `https://coverr.co/videos/morning-sunshine-Of9KYVXN1C`
};

it(`renders correctly`, () => {
  const tree = renderer.create(
      <SmallMovieCard film={movie} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`renders VideoPlayer component when the card is active`, () => {
  const id = 36;

  const tree = renderer.create(
      <SmallMovieCard film={movie} activeCardId={id} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`renders an image when the card is not active`, () => {
  const id = 4;

  const tree = renderer.create(
      <SmallMovieCard film={movie} activeCardId={id} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`renders an image when the card is not active`, () => {
  const id = null;

  const tree = renderer.create(
      <SmallMovieCard film={movie} activeCardId={id} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
