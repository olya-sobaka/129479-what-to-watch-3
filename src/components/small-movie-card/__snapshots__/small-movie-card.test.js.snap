// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders VideoPlayer component when the card is active 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseOut={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      className="player__video"
      height="175"
      muted={true}
      poster="img/midnight-special.jpg"
      width="280"
    >
      <source
        src="https://coverr.co/videos/norway-road-f0J1etTrcQ"
        type="video/mp4"
      />
      <source
        src="https://coverr.co/videos/morning-sunshine-Of9KYVXN1C"
        type="video/webm"
      />
    </video>
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
    >
      Kiki's Delivery Service
    </a>
  </h3>
</article>
`;

exports[`renders an image when the card is not active 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseOut={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Kiki's Delivery Service"
      height="175"
      src="img/midnight-special.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
    >
      Kiki's Delivery Service
    </a>
  </h3>
</article>
`;

exports[`renders an image when the card is not active 2`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseOut={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Kiki's Delivery Service"
      height="175"
      src="img/midnight-special.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
    >
      Kiki's Delivery Service
    </a>
  </h3>
</article>
`;

exports[`renders correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseOut={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Kiki's Delivery Service"
      height="175"
      src="img/midnight-special.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
    >
      Kiki's Delivery Service
    </a>
  </h3>
</article>
`;
